config/config.go

```diff
@@ -480,6 +504,29 @@ func RegisterInt64ConfigVariable(defaultValue int64, ptr *int64, isHotReloadable
 	}
 }
 
+func DeregisterConfigVariableWithPtr(ptr interface{}) {
+	configVarLock.Lock()
+	defer configVarLock.Unlock()
+
+	for key, configVars := range hotReloadableConfig {
+		for _, configvar := range configVars {
+			if configvar.value == ptr {
+				delete(hotReloadableConfig, key)
+				break
```

Comment:
Why do we break here instead of returning? Do we have a case where a pointer is both hotreloadable and not hotreloadable at the same time?

Reply:
Also, we are deleting the whole key... Shouldn't we just get rid of the pointer requested to be deleted?

Reply:
Based on the fact that a variable can be either hotreloadable or hotreloadable and not both, return makes sense. Changed.

Reply:
There is no point in keeping the config var in the map anymore after deregistering. So deleting it.